test/lib/linter/ui5Types/utils.ts

```diff
@@ -380,6 +381,243 @@ test("getPropertyAssignmentsInObjectLiteralExpression - unsupported elements", (
 	t.is(result[1], test2);
 });
 
+test("isReturnValueUsed (negative): Block", (t) => {
+	const program = createProgram(`
+		method();
+	`);
+	const expressionStatement = program.getSourceFile("test.ts")!.statements[0] as ts.ExpressionStatement;
+	t.false(isReturnValueUsed(expressionStatement));
+});
+
+test("isReturnValueUsed: CallExpression", (t) => {
+	const program = createProgram(`
+		foo(method());
+	`);
+	const expressionStatement = program.getSourceFile("test.ts")!.statements[0] as ts.ExpressionStatement;
+	const callExpression = expressionStatement.expression as ts.CallExpression;
+	const method = callExpression.arguments[0] as ts.CallExpression;
+	t.true(isReturnValueUsed(method));
+});
+
+test("isReturnValueUsed: BinaryExpression", (t) => {
+	const program = createProgram(`
+		true && method();
+	`);
+	const expressionStatement = program.getSourceFile("test.ts")!.statements[0] as ts.ExpressionStatement;
+	const binaryExpression = expressionStatement.expression as ts.BinaryExpression;
+	t.true(isReturnValueUsed(binaryExpression.right));
+});
+
+test("isReturnValueUsed: VariableDeclaration", (t) => {
+	const program = createProgram(`
+		const result = method();
+	`);
+	const variableStatement = program.getSourceFile("test.ts")!.statements[0] as ts.VariableStatement;
+	const variableDeclaration = variableStatement.declarationList.declarations[0];
+	t.true(isReturnValueUsed(variableDeclaration.initializer!));
+});
+
+test("isReturnValueUsed: IfStatement", (t) => {
+	const program = createProgram(`
+		if (method()) {}
+	`);
+	const ifStatement = program.getSourceFile("test.ts")!.statements[0] as ts.IfStatement;
+	t.true(isReturnValueUsed(ifStatement.expression));
+});
+
+test("isReturnValueUsed: ConditionalExpression", (t) => {
+	const program = createProgram(`
+		method() ? 1 : undefined;
+	`);
+	const expressionStatement = program.getSourceFile("test.ts")!.statements[0] as ts.ExpressionStatement;
+	const conditionalExpression = expressionStatement.expression as ts.ConditionalExpression;
+	t.true(isReturnValueUsed(conditionalExpression.condition));
+});
+
+test("isReturnValueUsed (negative): ConditionalExpression - whenTrue", (t) => {
+	const program = createProgram(`
+		condition ? method() : undefined;
+	`);
+	const expressionStatement = program.getSourceFile("test.ts")!.statements[0] as ts.ExpressionStatement;
+	const conditionalExpression = expressionStatement.expression as ts.ConditionalExpression;
+	t.false(isReturnValueUsed(conditionalExpression.whenTrue));
```

Comment:
I'm wondering whether we should consider this as `true`?
Why would you have a ternary operator and not expect a valuie from it?

Reply:
If the value is not used (void), then it doesn't matter.